File: JULY-2025/15-07-2025/p3136.ex
defmodule Solution do
  @spec is_valid(word :: String.t) :: boolean
  def is_valid(word) do
    chars = String.to_charlist(word)

    String.length(word) >= 3 and
        Enum.all?(chars,
            fn char -> (char in ?0..?9) or is_letter(char) end)
            and
            Enum.any?(chars, &is_vowel/1) and
            Enum.any?(chars, &(is_letter(&1) and not is_vowel(&1)))
  end

  def is_letter(char) do
    char in ?a..?z or char in ?A..?Z
  end

  def is_vowel(char) do
    char in [?a,?e,?i,?o,?u,?A,?E,?I,?O,?U]
  end
end

File: JULY-2025/15-07-2025/p3136.kt
class Solution {
    fun isValid(word: String): Boolean {
        if (word.length < 3) return false

        var vowels = 0
        var consonants = 0

        for (char in word) {
            if (isNotDigit(char) && isNotLetter(char)) return false
            if (isLetter(char)) {
                if (isVowel(char)) vowels++
                else consonants++
            }
        }

        return if (vowels == 0 || consonants == 0) false else true
    }

    private fun isNotDigit(char: Char) = when {
        char in '0'..'9' -> false
        else -> true
    }

    private fun isLetter(char: Char) = when {
        char in 'a'..'z' || char in 'A'..'Z' -> true
        else -> false
    }

    private fun isNotLetter(char: Char) = when {
        char in 'a'..'z' || char in 'A'..'Z' -> false
        else -> true
    }

    private fun isVowel(char: Char) = when {
        char in "aeiou" || char in "AEIOU" -> true
        else -> false
    }
}